internal/deviations/README.md

```diff
@@ -0,0 +1,88 @@
+## Guidelines to add deviations to FNT tests
+
+* Add the deviation to the `Deviation` message in the [proto/metadata.proto](https://github.com/openconfig/featureprofiles/blob/main/proto/metadata.proto) file.
+
+```
+message Deviations {
+...
+// Device does not support fragmentation bit for traceroute.
+bool traceroute_fragmentation = 2;
+...
+}
+```
+
+* Run `make proto/metadata_go_proto/metadata.pb.go` from your featureprofiles root directory to generate the Go code for the added proto fields.
+
+```
+$ make proto/metadata_go_proto/metadata.pb.go
+mkdir -p proto/metadata_go_proto
+# Set directory to hold symlink
+mkdir -p protobuf-import
+# Remove any existing symlinks & empty directories
+find protobuf-import -type l -delete
+find protobuf-import -type d -empty -delete
+# Download the required dependencies
+go mod download
+# Get ondatra modules we use and create required directory structure
+go list -f 'protobuf-import/{{ .Path }}' -m github.com/openconfig/ondatra | xargs -L1 dirname | sort | uniq | xargs mkdir -p
+go list -f '{{ .Dir }} protobuf-import/{{ .Path }}' -m github.com/openconfig/ondatra | xargs -L1 -- ln -s
+protoc -I='protobuf-import' --proto_path=proto --go_out=./ --go_opt=Mmetadata.proto=proto/metadata_go_proto metadata.proto
+goimports -w proto/metadata_go_proto/metadata.pb.go
+```
+
+* Add the accessor function for this deviation to the [internal/deviations/deviations.go](https://github.com/openconfig/featureprofiles/blob/main/internal/deviations/deviations.go) file. This function will need to accept a parameter `dut` of type `*ondatra.DUTDevice` to lookup the deviation value for a specific dut. This accessor function must call `lookupDUTDeviations` and return the deviation value. Test code will use this function to access deviations.
+	* If default value of deviation is same as default value for proto field, accessor method can directly call the `Get*()` method for deviation field. For e.g. for `bool` fields, the default is `false`, and the accessor method will have one line `return lookupDUTDeviations(dut).GetTracerouteFragmentation()`.
+
+	```
+	// TraceRouteFragmentation returns if the device does not support fragmentation bit for traceroute.
+	// Default value is false.
+	func TraceRouteFragmentation(dut *ondatra.DUTDevice) bool {
+	  return lookupDUTDeviations(dut).GetTracerouteFragmentation()
+	}
+	```
+
+	* If the default value of deviation is not the same as the default value of the proto field, the accessor method can add a check and return the required default value.
+
+	```
+	// HierarchicalWeightResolutionTolerance returns the allowed tolerance for BGP traffic flow while comparing for pass or fail conditions.
+	// Default minimum value is 0.2. Anything less than 0.2 will be set to 0.2.
+	func HierarchicalWeightResolutionTolerance(dut *ondatra.DUTDevice) float64 {
+    hwrt := lookupDUTDeviations(dut).GetHierarchicalWeightResolutionTolerance()
```

Comment:
The body of the function needs to be indented. Look at the preview to see how it's rendered:
https://github.com/openconfig/featureprofiles/blob/2fbd835c1654be786744edce1f9a4254b99ca397/internal/deviations/README.md

Reply:
done